File: task3.go
package main

import "fmt"

type Fraction struct {
	numerator   uint64
	denominator uint64
}

// НОД
func gcd(a, b uint64) uint64 {
	for b != 0 {
		temp := b
		b = a % b
		a = temp
	}

	return a
}

// сокращение дроби
func reduce(f Fraction) Fraction {
	g := gcd(f.numerator, f.denominator)

	f.numerator /= g
	f.denominator /= g

	return f
}

// сложение дробей
func add(a, b Fraction) Fraction {
	result := Fraction{
		numerator:   a.numerator*b.denominator + b.numerator*a.denominator,
		denominator: a.denominator * b.denominator,
	}

	return reduce(result)
}

// умножение дроби на целое число
func multiply(f Fraction, value uint64) Fraction {
	f.numerator *= value
	return reduce(f)
}

// деление дроби на целое число
func divide(f Fraction, value uint64) Fraction {
	f.denominator *= value
	return reduce(f)
}

// биномиальный коэффициент C(n, k)
func combination(n, k int) uint64 {
	var result uint64 = 1

	for i := 1; i <= k; i++ {
		result = result * uint64(n-i+1) / uint64(i)
	}

	return result
}

// вычисление суммы ряда
func calculateSum(a, b int) string {
	if b == 1 {
		return "infinity"
	}

	var sums [11]Fraction

	// сумма ряда 1 / b^n
	sums[0] = reduce(Fraction{
		numerator:   1,
		denominator: uint64(b - 1),
	})

	// вычисление сумм для степеней от 1 до a
	for k := 1; k <= a; k++ {
		current := Fraction{
			numerator:   1,
			denominator: 1,
		}

		for j := 0; j < k; j++ {
			term := multiply(sums[j], combination(k, j))
			current = add(current, term)
		}

		sums[k] = divide(current, uint64(b-1))
	}

	answer := reduce(sums[a])

	return fmt.Sprintf("%d/%d", answer.numerator, answer.denominator)
}

func main() {
	var a, b int

	fmt.Scan(&a, &b)

	if a < 1 || a > 10 || b < 1 || b > 10 {
		return
	}

	fmt.Println(calculateSum(a, b))
}

File: task3.cpp
#include <iostream>
#include <string>
#include <cstdint>

using namespace std;

struct Fraction {
    uint64_t numerator;
    uint64_t denominator;
};

// НОД
uint64_t gcd(uint64_t a, uint64_t b) {
    while (b != 0) {
        uint64_t temp = b;
        b = a % b;
        a = temp;
    }

    return a;
}

// сокращение дроби
Fraction reduce(Fraction f) {
    uint64_t g = gcd(f.numerator, f.denominator);

    f.numerator /= g;
    f.denominator /= g;

    return f;
}

// сложение дробей
Fraction add(Fraction a, Fraction b) {
    Fraction result;

    result.numerator =
        a.numerator * b.denominator +
        b.numerator * a.denominator;

    result.denominator =
        a.denominator * b.denominator;

    return reduce(result);
}

// умножение дроби на целое число
Fraction multiply(Fraction f, uint64_t value) {
    f.numerator *= value;
    return reduce(f);
}

// деление дроби на целое число
Fraction divide(Fraction f, uint64_t value) {
    f.denominator *= value;
    return reduce(f);
}

// биномиальный коэффициент C(n, k)
uint64_t combination(int n, int k) {
    uint64_t result = 1;

    for (int i = 1; i <= k; i++) {
        result = result * (n - i + 1) / i;
    }

    return result;
}

// вычисление суммы ряда
string calculateSum(int a, int b) {
    if (b == 1) {
        return "infinity";
    }

    Fraction sums[11];

    // сумма ряда 1 / b^n
    sums[0] = reduce({1, static_cast<uint64_t>(b - 1)});

    // вычисление сумм для степеней от 1 до a
    for (int k = 1; k <= a; k++) {
        Fraction current = {1, 1};

        for (int j = 0; j < k; j++) {
            Fraction term = multiply(sums[j], combination(k, j));
            current = add(current, term);
        }

        sums[k] = divide(current, b - 1);
    }

    Fraction answer = reduce(sums[a]);

    return to_string(answer.numerator) + "/" + to_string(answer.denominator);
}

int main() {
    int a, b;

    cout << "Введите a и b: ";
    cin >> a >> b;

    if (cin.fail()) {
        return 0;
    }

    if (a < 1 || a > 10 || b < 1 || b > 10) {
        return 0;
    }

    cout << "Результат: " << calculateSum(a, b) << endl;

    return 0;
}
